src/common/lint/add-lint-configs.ts

```diff
@@ -1,11 +1,42 @@
+import type {Linter} from 'eslint'
+import {JsonFile} from 'projen'
 import {NodeProject} from 'projen/lib/javascript'
+import {deepMerge} from 'projen/lib/util'
+import {eslintConfigFormatting} from './configs/eslint-config-formatting'
+import {eslintConfigQuality} from './configs/eslint-config-quality'
+import {prettierConfig} from './configs/prettier'
 
-export const addLintConfigs = (project: NodeProject, extraEslintConfigs: Array<string> = []): void => {
-  project.package.addField('prettier', '@ottofeller/prettier-config-ofmt')
+const arrayProperties = ['extends', 'overrides', 'plugins'] as const
+type LinterConfigArrayProperties = Pick<Linter.Config, typeof arrayProperties[number]>
 
+export const addLintConfigs = (project: NodeProject, extraEslintConfigs: Array<Linter.Config>): void => {
+  // ANCHOR Prettier
+  new JsonFile(project, '.prettierrc.json', {obj: prettierConfig, marker: false})
+
+  // ANCHOR Eslint
   const commonConfigs = project.parent
     ? [] // NOTE For a subproject we assume the parent is created from one of our templates and thus has the configs.
-    : ['@ottofeller/eslint-config-ofmt/eslint.quality.cjs', '@ottofeller/eslint-config-ofmt/eslint.formatting.cjs']
+    : [eslintConfigFormatting, eslintConfigQuality]
+
+  const allConfigs = [...commonConfigs, ...extraEslintConfigs]
+
+  if (allConfigs.length === 0) {
+    return
+  }
+
+  const mergedArrayProperties = arrayProperties.reduce<LinterConfigArrayProperties>((accumulator, prop) => {
```

Comment:
`deepMerge` from _projen_ does not merge arrays. Thus we need to merge them manually. Here only a needed subset is handled, a general case might be added later if really needed.